### ISSUE TITLE
Is it possible to build this library under msys2's gcc?

### ISSUE THREAD

**Issue by asmwarrior** [kovacsv/VisualScriptEngine#14]
I looked at the cmakelist file, this file mainly are support the Microsoft visual studio c++, but not mingw gcc?

**Comment by kovacsv** [comment]
The engine is tested with several compilers, see the https://github.com/kovacsv/VisualScriptEngine/wiki/Building-the-Engine page for details.

I think it should work with mingw gcc, too. Juse generate makefiles with cmake, and use the gcc compiler to build.

**Comment by asmwarrior** [comment]
First, thanks for the reply.

But if you look at the CMakeLists.txt, I see some code snippet:

~~~~
function (SetCompilerOptions module)
	target_compile_options (${module} PUBLIC "$<$<CONFIG:Debug>:-DDEBUG>")
	if (WIN32)
		set (AdditionalWarnings /w44061 /w44062 /w44265 /w44266 /w44355 /w44596 /w44800)
		target_compile_options (${module} PUBLIC /W4 /WX ${AdditionalWarnings} ${VSE_CUSTOM_BUILD_OPTIONS})
	else ()
		target_compile_options (${module} PUBLIC -std=c++11 -Wall -Wextra -Werror ${VSE_CUSTOM_BUILD_OPTIONS})
	endif ()
endfunction ()
~~~~

This looks like under Windows, it has only support the visual studio C++ compiler options. `/w` like option is only for this compiler.

Thanks.

**Comment by kovacsv** [comment]
Yes, you are right, instead of `WIN32`, it should check visual studio. Have you tried to delete the `WIN32` branch, and use the other one for mingw?

**Comment by asmwarrior** [comment]
> Yes, you are right, instead of `WIN32`, it should check visual studio. Have you tried to delete the `WIN32` branch, and use the other one for mingw?

I'm not familiar with cmake, so I don't know how to tweak the cmakelist file.

BTW:
What does the "visual script" used for, from the demo https://kovacsv.github.io/VisualScriptEngineWebDemo/
I see that it can do some simple calculation (show the parsing tree of the calculator)?

Thanks.

**Comment by kovacsv** [comment]
Please try to modify the function in the `CMakeLists.txt` file like this, and see if it works with mingw.

```
function (SetCompilerOptions module)
	target_compile_options (${module} PUBLIC "$<$<CONFIG:Debug>:-DDEBUG>")
	target_compile_options (${module} PUBLIC -std=c++11 -Wall -Wextra -Werror ${VSE_CUSTOM_BUILD_OPTIONS})
endfunction ()
```

**Comment by zzc-1024** [comment]
Hi, I tried to fix codes. Eventually, it can generate executable files. However, when I opened it, it show me "Unable to locate program input point _ZNKSt19_codecvt_utf8_baselwE10do_unshiftER9_MbstatetPcS3_RS3_"
![image](https://user-images.githubusercontent.com/67462033/221086777-035c20c8-fdb6-482f-9776-01fe6c05da47.png)

**Comment by asmwarrior** [comment]
> Hi, I tried to fix codes. Eventually, it can generate executable files. However, when I opened it, it show me "Unable to locate program input point _ZNKSt19_codecvt_utf8_baselwE10do_unshiftER9_MbstatetPcS3_RS3_" ![image](https://user-images.githubusercontent.com/67462033/221086777-035c20c8-fdb6-482f-9776-01fe6c05da47.png)

I think you need to use some dependency check tool to check which dll files are used by the exe. Maybe, wrong DLL file is used.

**Comment by zzc-1024** [comment]
> > Hi, I tried to fix codes. Eventually, it can generate executable files. However, when I opened it, it show me "Unable to locate program input point _ZNKSt19_codecvt_utf8_baselwE10do_unshiftER9_MbstatetPcS3_RS3_" ![image](https://user-images.githubusercontent.com/67462033/221086777-035c20c8-fdb6-482f-9776-01fe6c05da47.png)
> 
> I think you need to use some dependency check tool to check which dll files are used by the exe. Maybe, wrong DLL file is used.

Thank you. After dependency check, I found that program missing libstdc++-6.lib. The program can run now.
![image](https://user-images.githubusercontent.com/67462033/221106769-38afdec4-db57-4b18-820f-2f5a5353bbd4.png)
